### PR TITLE
Feature/storages creation

### PR DESCRIPTION
**Pull request by 8r1m2t0n3**

### CODE REVIEW

File: src/main/java/org/andersen/homework/util/collection/MyHashSet.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package org.andersen.homework.util.collection;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.stream.Collectors;
+
+public class MyHashSet<E> {
+
+  private HashMap<E, Object> map;
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/util/collection/MyHashSet.java]: Don't use here HashMap, instead use array, ArrayList ot LinkedList

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+package org.andersen.homework.util.collection;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class MyArrayList<E> {
+
+  private static final Object[] DEFAULT_EMPTY_CAPACITY_ARRAY = {};
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/util/collection/MyArrayList.java:10]: Why this field is static?

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+  private Object[] array;
+
+  private int capacity;
+
+  private int entriesCount = 0;
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/util/collection/MyArrayList.java]: do we need declaration here?)

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    } else if (initialCapacity == 0) {
+      this.array = DEFAULT_EMPTY_CAPACITY_ARRAY;
+    } else {
+      throw new IllegalArgumentException("Illegal Capacity: " + initialCapacity);
+    }
+
+    this.capacity = initialCapacity;
+  }
+
+  @SuppressWarnings("unchecked")
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/util/collection/MyArrayList.java:35]: Why do we need here this annotation?

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  public E get(int index) {
+    Objects.checkIndex(index, entriesCount);
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/util/collection/MyArrayList.java]: What does this method do?)

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class MyArrayList<E> {
+
+  private static final Object[] DEFAULT_EMPTY_CAPACITY_ARRAY = {};
+
+  private Object[] array;
+
+  private int capacity;
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/util/collection/MyArrayList.java:14]: Why do we need this field?

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java (hunk 4 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/util/collection/MyArrayList.java:35]: Code below unchecked casting type E to Object and compiler shows warning. This annotation disables the warning

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java (hunk 2 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/util/collection/MyArrayList.java:10]: I've made it static just to  mark that it's an already set constant related to a class, not to specific instance

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java (hunk 5 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/util/collection/MyArrayList.java]: Checks if 'index' is out of bounds, throws exception if so

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java (hunk 6 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/util/collection/MyArrayList.java:14]: This field contains amount of allocated memory for the 'array' field. Number of elements in an array may change, but it's capacity - never

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java (hunk 3 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/util/collection/MyArrayList.java]: This field represents number of elements in 'array', but not a size of it

Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/util/collection/MyArrayList.java]: Going to rename it to 'elementsCount'

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    this.capacity = initialCapacity;
+  }
+
+  @SuppressWarnings("unchecked")
+  public E get(int index) {
+    Objects.checkIndex(index, elementsCount);
+    if (index > elementsCount - 1) {
+      return null;
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/util/collection/MyArrayList.java]: In which case we return null? Could you provide test for this in Main class?(just sout the value after getting )

**Review by Darya-Drutsko — APPROVED** [review]

File: src/main/java/org/andersen/homework/util/collection/MyArrayList.java (hunk 7 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/util/collection/MyArrayList.java]: You're right, this statement is redundant. `Objects.checkIndex(index, elementsCount)` already does all the work, seems like I just forget to remove it after having a better realization idea

**Pull request merged**